test/python/compiler/test_assembler.py

```diff
@@ -569,6 +586,32 @@ def test_assemble_meas_map(self):
         with self.assertRaises(QiskitError):
             assemble(schedule, meas_map=[[0, 1, 2]])
 
+    def test_assemble_memory_slots(self):
```

Comment:
This will be deprecated soon, so there is no real need to add this test. Additionally, this test is about assembly and not the DeviceSpecification.